driver-core/src/main/com/mongodb/client/model/geojson/Geometry.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.geojson;
+
+import com.mongodb.client.model.geojson.codecs.GeoJsonCodecProvider;
+import org.bson.codecs.Codec;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistries;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.json.JsonWriter;
+import org.bson.json.JsonWriterSettings;
+
+import java.io.StringWriter;
+
+/**
+ * An abstract class for representations of GeoJSON geometry objects.
+ *
+ * @since 3.1
+ */
+public abstract class Geometry {
```

Comment:
1. I considered it, but decided against it, as there are no other classes named Point, Position, etc that are likely to be used in a server application.
2. Just because it can be added if and when we add features support, without breaking binary compatibility.